mlflow/cli.py

```diff
@@ -129,11 +132,11 @@ def run(uri, entry_point, version, param_list, experiment_name, experiment_id, b
             experiment_id=experiment_id,
             parameters=param_dict,
             backend=backend,
-            backend_config=cluster_spec_arg,
+            backend_config=backend_config,
             use_conda=(not no_conda),
             storage_dir=storage_dir,
-            synchronous=backend == "local" or backend is None,
-            run_id=run_id,
+            synchronous=backend in ("local", "kubernetes") or backend is None,
```

Comment:
Should we make it async by default?

Reply:
I think it would be better this way, so we can monitoring the execution easier. The other way would be to grab info from kubernetes with kubectl or monitor mlflow ui to see when the run had finished. What do you think?

Reply:
@tomasatdatabricks and @andrewmchen , what do you guys think about get pod's log in case of errors? I'm reviewing @tomasatdatabricks 's changes for job monitoring and I'm getting some errors from pods execution. Now I need to run `kubectl` to find the pod related to the job and get it's logs.

We could ease the debugging if we just print out the logs in case of errors. We can chat on Slack if you like.

Reply:
Hey @marcusrehm. 

That is a good idea. But I think the current PR provides good functionality as is and we can leave this for a follow up PR.

It's not 100% clear what is the right thing to do regarding the logs. For example, we might also pull them after every run and attach them to the run as artifact(s).  Streaming the output as you had it in your PR may also be a good idea. It may be confusing thought in case multiple pods run during the execution (either due to parallelism or retry).  .

Reply:
Hey @tomasatdatabricks , I think you're right. We can let it for another PR. I've tested the logic you changed and it seams okay!

If you and @andrewmchen think we have nothing left to change then I think we can close the PR.

Reply:
The build is broken but it's related to spark testing, not to what we are doing.

Reply:
I'll take a look. Maybe we just need to rerun.

Reply:
Thanks @tomasatdatabricks !